cpp/src/arrow/python/builtin_convert.cc

```diff
@@ -830,38 +821,51 @@ Status AppendPySequence(PyObject* obj, int64_t size,
   return converter->AppendData(obj, size);
 }
 
-Status ConvertPySequence(PyObject* obj, MemoryPool* pool, std::shared_ptr<Array>* out) {
+static Status ConvertPySequenceReal(PyObject* obj, MemoryPool* pool,
+                                    std::shared_ptr<Array>* out,
+                                    const std::shared_ptr<DataType>* type, int64_t size) {
```

Comment:
This `size` argument should precede the "out" arguments (https://google.github.io/styleguide/cppguide.html#Function_Parameter_Ordering)